src/matchers/has.ts

```diff
@@ -1,17 +1,27 @@
 // Dependencies:
 import { Node } from 'typescript';
 import { findMatches } from '../match';
-import { traverseChildren } from '../traverse';
+import { traverse } from '../traverse';
 import { TSQueryOptions, TSQuerySelectorNode } from '../tsquery-types';
 
-export function has (node: Node, selector: TSQuerySelectorNode, _: Array<Node>, options: TSQueryOptions): boolean {
+export function has (node: Node, selector: TSQuerySelectorNode, ancestry: Array<Node>, {}: Node, {}: TSQueryOptions): boolean {
```

Comment:
I got the `:scope` selector by moving the `traverseChildren` function in here. It's probably not the best way to do it.